GSoC_19/posts/websocket-passive-scan-using-scripts-with-zap.org

```diff
@@ -0,0 +1,851 @@
+    it. See what alert was raised. 
+
+    Finally: Go to Alerts Panel hit the broom icon to remove all Alerts.
+    
+** Script Console Add-on
+
+   And now one of my favorite features of ZAP, the Script Console add-on.
+
+   #+BEGIN_QUOTE
+   The Script Console add-on allows you to run scripts that can be embedded within ZAP and can access internal ZAP data structures. It supports any scripting language that supports JSR 223 (http://www.jcp.org/en/jsr/detail?id=223) , including:
+
+   ECMAScript / Javascript (using Nashorn engine, included by default)
+
+   Zest https://developer.mozilla.org/en-US/docs/zest (included by default)
+
+   Groovy http://groovy-lang.org/
+
+   Python http://www.jython.org
+
+   Ruby - http://jruby.org/
+
+   and many more...
+   
+   /from [[https://github.com/zaproxy/zap-core-help/wiki/HelpAddonsScriptsScripts][wiki]]/
+   #+END_QUOTE
+
+   Referring to scripts console add-on we are referring to the ability that the
+   user has to write scripts which implement various behaviors like scans, fuzzing etc.
+   Scripts Console add-on has a panel where the scripts are displayed. The panel
+   has a tree structure, which has two branches, the =Scripts= branch and
+   =Templates=. In both branches there are some folders which are referring on
+   different kind of scripts. As you can see those folders in both branches have
+   the same names.  
+
+   #+ATTR_HTML: :align center :width 300 height 800
+   #+CAPTION: Scripts Tree Sidebar
+   [[../../images/pscan-tutorial/scripts_tree.png]]
+
+   ZAP defines =Templates= in order to be used as base scripts. As we will see in
+   a while every kind of script should implement a specific functionality, named
+   rule. In addition, =Templates= are not runnable, enablable or directly
+   editable, they are used only as templates. In order to use a template, right
+   click on it and then hit =New script...=. 
+
+   **Hands On Section:**
+   Go to the template script =Base64 Disclosure.js=, right click on it, change
+   =Script Name= to =MyBase64 Disclosure.js= and hit save. See what happened.
+
+   =Scripts=, on the other hand, are the actual scripts which are run by ZAP.
+   There are a good many possibilities in which we can use scripts,
+   however, in this guide we are interested in WebSocket Passive Scanning.
+   Passive Scripts have two states, enable and disable. When a script is disabled means
+   that it is not running at all. Other options like =duplicate= script or =remove= script are
+   available by right clicking on them. **You should always remember that Right**
+   **Click in ZAP is your best friend.** By default, the folders under =Scripts=
+   are empty, except from the =WebSocket Passive Rules=. If you want to explore
+   more the scripting techniques and add some rules in ZAP, I totally recommend
+   to install =Community Scripts= from =marketplace=. 
```

Comment:
Include  a note here (or somewhere) that the scripts will be automatically disabled if an exception is encountered?